backend/src/main/java/de/neuefische/CapStone/backend/service/AuthService.java

```diff
@@ -0,0 +1,56 @@
+package de.neuefische.CapStone.backend.service;
+
+import de.neuefische.CapStone.backend.model.UserEntity;
+import de.neuefische.CapStone.backend.repo.UserRepository;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityExistsException;
+import java.util.Optional;
+
+import static org.springframework.util.StringUtils.hasText;
+
+@Service
+@Getter
+@Setter
+public class AuthService {
+
+    private UserRepository userRepository;
+
+    @Autowired
+    public AuthService(UserRepository userRepository){
+        this.userRepository=userRepository;
+    }
+
+    public UserEntity createUser(UserEntity userEntity) {
+        String name = userEntity.getUserName();
+        if(!hasText(name)) {
+            throw new IllegalArgumentException("Name must not be blank");
+        }
+        checkIfUserExists(name);
+        String email = userEntity.getEmail();
+        if(!hasText(email)) {
+            throw new IllegalArgumentException("Email must not be blank");
+        }
+        checkIfEmailExists(email);
+        userRepository.save(userEntity);
+        return userEntity;
+
+    }
+
+    private void checkIfEmailExists(String email) {
+        Optional<UserEntity> existingEmail = userRepository.findByEmail(email);
```

Comment:
Maybe you could use a `existsByEmail` query method to the repository?

Reply:
> Good job and easy to review 👍
> 
> I am wondering about the docker stuff in this pr..?

Thanks for the review, I implemented all of those thing. The Docker Stuff was part of my "Oh no, I don´t know why my database isn´t working"-reaction, but its necessary (I think). I should have added that into the PR-comment though.